mobile/src/main/java/org/openhab/habdroid/ui/homescreenwidget/ItemUpdateWidget.kt

```diff
@@ -0,0 +1,234 @@
+import org.openhab.habdroid.ui.AbstractItemPickerActivity
+import org.openhab.habdroid.ui.PreferencesActivity
+import org.openhab.habdroid.util.HttpClient
+import org.openhab.habdroid.util.bitmapToSvg
+import org.openhab.habdroid.util.dpToPixel
+import org.openhab.habdroid.util.getIconFormat
+import org.openhab.habdroid.util.getPrefs
+import org.openhab.habdroid.util.getString
+import org.openhab.habdroid.util.isIconFormatPng
+import java.io.IOException
+import kotlin.math.min
+
+open class ItemUpdateWidget : AppWidgetProvider() {
+    override fun onUpdate(context: Context, appWidgetManager: AppWidgetManager, appWidgetIds: IntArray) {
+        Log.d(TAG, "onUpdate()")
+        appWidgetIds.forEach {
+            Log.d(TAG, "id: $it")
+            setupWidget(context, getInfoForWidget(context, it), it, appWidgetManager)
+        }
+        super.onUpdate(context, appWidgetManager, appWidgetIds)
+    }
+
+    override fun onAppWidgetOptionsChanged(
+        context: Context?,
+        appWidgetManager: AppWidgetManager?,
+        appWidgetId: Int,
+        newOptions: Bundle?
+    ) {
+        Log.d(TAG, "onAppWidgetOptionsChanged()")
+        val data = getInfoForWidget(context ?: return, appWidgetId)
+        setupWidget(context, data, appWidgetId, appWidgetManager ?: return)
+    }
+
+    override fun onReceive(context: Context?, intent: Intent?) {
+        Log.d(TAG, "onReceive() $intent")
+        super.onReceive(context, intent)
+    }
+
+    override fun onDeleted(context: Context?, appWidgetIds: IntArray?) {
+        Log.d(TAG, "onDeleted()")
+        if (context == null || appWidgetIds == null) {
+            return
+        }
+        appWidgetIds.forEach {
+            Log.d(TAG, "Deleting data for id $it")
+            context.deleteFile(getFileNameForWidget(it))
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                context.deleteSharedPreferences(getFileNameForWidget(it))
+            } else {
+                context.getSharedPreferences(getFileNameForWidget(it), MODE_PRIVATE).edit {
+                    clear()
+                }
+            }
+        }
+        super.onDeleted(context, appWidgetIds)
+    }
+
+    companion object {
+        private val TAG = ItemUpdateWidget::class.java.simpleName
+
+        fun setupWidget(
+            context: Context,
+            data: ItemUpdateWidgetData,
+            appWidgetId: Int,
+            appWidgetManager: AppWidgetManager
+        ) {
+            val widgetOptions = appWidgetManager.getAppWidgetOptions(appWidgetId)
+            val smallWidget = widgetOptions.getInt(AppWidgetManager.OPTION_APPWIDGET_MAX_HEIGHT) <
+                context.resources.getDimension(R.dimen.small_widget_threshold)
+            val views = RemoteViews(context.packageName,
+                if (smallWidget) R.layout.widget_item_update_small else R.layout.widget_item_update)
+            val intent = Intent(context, BackgroundIntentReceiveActivity::class.java).apply {
```

Comment:
Widgets should just send their id to `ItemUpdateWidget` and `onReceive()` should handle the intent.

Reply:
Done